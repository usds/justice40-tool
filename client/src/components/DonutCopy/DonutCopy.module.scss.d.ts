declare namespace DonutCopyNamespace {
    export interface IDonutCopyScss {
      donutCopyContainer: string;
      donutRow: string;
      donutRowLabel: string;
      invert: string;
      noInvert: string;
      valueSubTextContainer: string;
      subTextContainer: string;
    }
  }

declare const DonutCopyScssModule: DonutCopyNamespace.IDonutCopyScss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: DonutCopyNamespace.IDonutCopyScss;
  };

  export = DonutCopyScssModule;
