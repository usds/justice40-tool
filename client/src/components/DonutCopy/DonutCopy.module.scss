@use '../../styles/design-system.scss' as *;
@import "../utils.scss";

.donutCopyContainer{
    @include u-display('flex');
    flex-direction: column;
    @include u-padding-left(2);
    @include u-padding-right(2.5);
    @include u-padding-top(2);

    .donutRow {
        @include u-display('flex');
        justify-content: space-between;

        .donutRowLabel {
            font-weight: bold;
        }

        .valueSubTextContainer {
            @include u-display('flex');
            flex-direction: column;
            .subTextContainer{
                @include indicatorValueSubTextContainer();
            }
        }
    }

    .donutRow:first-child {
        @include u-padding-bottom("05");
    }

    .invert {
        align-self: flex-end;
        @include invert();
    }
    
    .noInvert {
        align-self: flex-end;
    }
}
