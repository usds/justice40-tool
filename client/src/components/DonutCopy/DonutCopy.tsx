import React from 'react';
import {useIntl} from 'gatsby-plugin-intl';

import {IndicatorValue, IndicatorValueSubText} from '../Indicator/Indicator';

import * as styles from './DonutCopy.module.scss';

import * as EXPLORE_COPY from '../../data/copy/explore';

export interface IDonutCopyProps {
  isAdjacent: boolean
  povertyBelow200Percentile: number | null
}

const DonutCopy = ({isAdjacent, povertyBelow200Percentile}: IDonutCopyProps) => {
  const intl = useIntl();
  const povBel200Percentile = povertyBelow200Percentile ?
    parseFloat((povertyBelow200Percentile*100).toFixed()) : null;
  const threshold = 50;

  return (
    <div className={styles.donutCopyContainer}>
      <div className={styles.donutRow}>
        <div className={styles.donutRowLabel}>{intl.formatMessage(EXPLORE_COPY.DONUT_COPY.COMP_SURR)}</div>
        <div className={isAdjacent ? styles.invert : ''}>
          {isAdjacent ? EXPLORE_COPY.COMMUNITY.OF_FOCUS : EXPLORE_COPY.COMMUNITY.NOT_OF_FOCUS}
        </div>
      </div>
      <div className={styles.donutRow}>
        <div className={styles.donutRowLabel}>{intl.formatMessage(EXPLORE_COPY.DONUT_COPY.ADJ_LOW_INC)}</div>
        <div className={styles.valueSubTextContainer}>
          <div className={
            isAdjacent && povBel200Percentile &&
            povBel200Percentile >= threshold ? styles.invert : styles.noInvert}>
            <IndicatorValue
              type={'percentile'}
              displayStat={povBel200Percentile}
            />
          </div>
          <div className={styles.subTextContainer}>
            <IndicatorValueSubText
              value={povBel200Percentile}
              isAboveThresh={povBel200Percentile && povBel200Percentile >= threshold ? true : false}
              threshold={threshold}
              type={'percentile'}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default DonutCopy;
