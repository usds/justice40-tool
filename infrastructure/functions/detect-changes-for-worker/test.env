REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1