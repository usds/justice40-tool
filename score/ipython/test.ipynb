{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
