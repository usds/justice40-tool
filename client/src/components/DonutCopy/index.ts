import DonutCopy from './DonutCopy';
export default DonutCopy;
