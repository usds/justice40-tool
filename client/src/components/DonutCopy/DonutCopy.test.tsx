import React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import DonutCopy from './DonutCopy';

describe('rendering of DonutCopy Component', () => {
  it('checks if component renders when adjacency is false', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <DonutCopy
            isAdjacent={false}
            povertyBelow200Percentile={32}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('checks if component renders when adjacency is false', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <DonutCopy
            isAdjacent={true}
            povertyBelow200Percentile={32}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
