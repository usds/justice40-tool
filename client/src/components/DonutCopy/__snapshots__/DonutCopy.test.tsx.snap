// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of DonutCopy Component checks if component renders when adjacency is false 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        Completely surrounded
      </div>
      <div
        class=""
      >
        NO
      </div>
    </div>
    <div>
      <div>
        Adjusted low income
      </div>
      <div>
        <div>
          3
          <sup
            style="top: -0.2em;"
          >
            th
          </sup>
        </div>
        <div>
          <div>
            above 50
            <sup
              style="top: -0.2em;"
            >
              th
            </sup>
             percentile
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering of DonutCopy Component checks if component renders when adjacency is false 2`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        Completely surrounded
      </div>
      <div>
        YES
      </div>
    </div>
    <div>
      <div>
        Adjusted low income
      </div>
      <div>
        <div>
          3
          <sup
            style="top: -0.2em;"
          >
            th
          </sup>
        </div>
        <div>
          <div>
            above 50
            <sup
              style="top: -0.2em;"
            >
              th
            </sup>
             percentile
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
